litex/tools/litex_json2dts.py

```diff
@@ -192,6 +195,34 @@ def generate_dts(d, initrd_start=None, initrd_size=None, polling=False):
     uart_csr_base  = d["csr_bases"]["uart"],
     uart_interrupt = "" if polling else "interrupts = <{}>;".format(d["constants"]["uart_interrupt"]))
 
+    if "uart1" in d["csr_bases"]:
+        aliases["serial1"] = "liteuart1"
+        dts += """
+            liteuart1: serial@{uart_csr_base:x} {{
+                device_type = "serial";
+                compatible = "litex,liteuart";
+                reg = <0x{uart_csr_base:x} 0x100>;
+                {uart_interrupt}
+                status = "okay";
+            }};
+""".format(
+    uart_csr_base  = d["csr_bases"]["uart1"],
+    uart_interrupt = "" if polling else "interrupts = <{}>;".format(d["constants"]["uart1_interrupt"]))
+
+    if "uart2" in d["csr_bases"]:
+        aliases["serial2"] = "liteuart2"
+        dts += """
+            liteuart2: serial@{uart_csr_base:x} {{
+                device_type = "serial";
+                compatible = "litex,liteuart";
+                reg = <0x{uart_csr_base:x} 0x100>;
+                {uart_interrupt}
+                status = "okay";
+            }};
+""".format(
+    uart_csr_base  = d["csr_bases"]["uart2"],
+    uart_interrupt = "" if polling else "interrupts = <{}>;".format(d["constants"]["uart2_interrupt"]))
+
```

Comment:
We are discussing this in https://github.com/enjoy-digital/litex/pull/777, for now it's still not decided if multiple instance of the same peripheral will be directly support in LiteX and preliminary work as described in #777 is required for that.